R/generateFilterValues.R

```diff
@@ -55,15 +55,16 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
 
   # ensemble
   if (class(method) == "list") {
-    ens.method = method[[1]]
-    method = method[[2]]
-    assertSubset(ens.method, choices = ls(.FilterEnsembleRegister), empty.ok = FALSE)
-    if (length(method) == 1) {
-      warningf("You only passed one base filter method to an ensemble filter. Please use at least two base filter methods to have a voting effect.")
+    if (method[[1]] %in% ls(.FilterEnsembleRegister)) {
+			ens.method = method[[1]]
+			method = method[[2]]
+			if (length(method) == 1) {
+				warningf("You only passed one base filter method to an ensemble filter. Please use at least two base filter methods to have a voting effect.")
+			}
     }
   }
 
-  assertSubset(method, choices = append(ls(.FilterRegister), ls(.FilterEnsembleRegister)), empty.ok = FALSE)
+  lapply(method, assertSubset, choices = append(ls(.FilterRegister), ls(.FilterEnsembleRegister)), empty.ok = FALSE)  
```

Comment:
would it be better to unlist and then take the old approach rather than using `lapply()` here?

Also doesn't `lapply()` trigger console output here since the result is not assigned to an object?

Also note the superfluous white-spaces at the end.

Reply:
OK. 
Note the call to lapply a few lines below that, which is not my code. Won't that trigger console output as well?
```
  if (!(any(sapply(filter, function(x) !isScalarNA(filter$pkg))))) {
    lapply(filter, function(x) requirePackages(x$pkg, why = "generateFilterValuesData", default.method = "load"))
```

Reply:
Looks like it, yes. Feel free to also fix it, maybe even just by assigning the outcome to an object. We have no side-effect helper funs such as `walk` in _mlr_ unfortunately.

Reply:
OK done.